src/client/sandbox/shadow-ui.js

```diff
@@ -131,15 +134,18 @@ export default class ShadowUI extends SandboxBase {
                     if (typeof args[0] === 'string')
                         args[0] = NodeSandbox.processSelector(args[0]);
 
-                    return sandbox._filterNodeList(nativeMethods[nativeQuerySelectorAllFnName].apply(this, args));
+                    const list = nativeMethods[nativeQuerySelectorAllFnName].apply(this, args);
+                    const length = nativeMethods.nodeListLengthGetter.call(list);
+
+                    return sandbox._filterNodeList(list, length);
                 };
             }
         };
     }
 
     _markShadowUIContainerAndCollections (containerEl) {
         nativeMethods.objectDefineProperty.call(this.window.Object, containerEl, IS_SHADOW_CONTAINER_FLAG, { value: true });
-        nativeMethods.objectDefineProperty.call(this.window.Object, containerEl.children, IS_SHADOW_CONTAINER_COLLECTION_FLAG, { value: true });
+        ShadowUI.markCollectionAsShadow(containerEl.children);
```

Comment:
Let's create the `ShadowUI.markAsShadowUIContainer` too and replace with it in all appropriate places.